TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/Book.kt

```diff
@@ -0,0 +1,202 @@
+import android.widget.LinearLayout
+import android.widget.TextView
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import kotlin.math.max
+import kotlin.math.min
+
+class Book(private val filePath: String, private val activityContext: Context) {
+    private val assetManager: AssetManager = activityContext.assets
+    private var chapterStarts: ArrayList<Int> = ArrayList()
+    private var chapterLengths: ArrayList<Int> = ArrayList()
+    val numChapters
+        get() = chapterStarts.size
+    var chapterStartString = activityContext.resources.getString(R.string.chapter_start)
+    var chapterEndString = activityContext.resources.getString(R.string.chapter_end)
+
+    init {
+        val bufferedReader = BufferedReader(InputStreamReader(assetManager.open(filePath)))
+
+        var line = 0
+        var lineCount = 0
+        var lineString: String? = ""
+        var emptyLineCount = 2
+
+        do {
+            if (lineString!!.isEmpty()) {
+                emptyLineCount += 1
+            } else {
+                if (emptyLineCount >= 3) {
+                    chapterStarts.add(line)
+                    chapterLengths.add(lineCount)
+                    lineCount = 0
+                }
+                emptyLineCount = 0
+            }
+            line += 1
+            lineCount += 1
+            lineString = bufferedReader.readLine()
+        } while (lineString != null)
+        bufferedReader.close()
+
+        chapterLengths.add(lineCount)
+        chapterLengths.removeAt(0)
+    }
+
+    private lateinit var paragraphStrings: ArrayList<String>
+    val chapterTitle
+        get() = "(Section ${currentChapter}) ${paragraphStrings[0]}"
+
+    var currentChapter = 0
+        set(inChapter) {
+            field = inChapter
+
+            paragraphStrings = ArrayList()
+
+            if (field >= 0 && field < chapterStarts.size) {
+                val bufferedReader = BufferedReader(InputStreamReader(assetManager.open(filePath)))
+
+                for (line in 0 until chapterStarts[field] - 1) {
+                    bufferedReader.readLine()
+                }
+
+                var paragraphBuffer = StringBuffer()
+                var lineBuffer: String
+                for (line in 0 until chapterLengths[field]) {
+                    lineBuffer = bufferedReader.readLine()
+                    if (lineBuffer.isEmpty()) {
+                        if (!paragraphBuffer.isEmpty()) {
+                            paragraphStrings.add(textStringHelper(paragraphBuffer.toString()))
+                            paragraphBuffer = StringBuffer()
+                        }
+                    } else {
+                        paragraphBuffer.append("$lineBuffer ")
+                    }
+                }
+                bufferedReader.close()
+            }
+
+            buildPages()
+        }
+
+
+    private var currentParagraph = 0
+
+    private lateinit var pageStrings: ArrayList<ArrayList<String>>
```

Comment:
Here and above, please, move all variables together at the top of the class and not mixed between functions along the class.

Reply:
Done